content/learning-paths/embedded-systems/restrict-keyword-c99/restrict-example-sve2.md

```diff
@@ -85,7 +85,7 @@ process_data:
         ret
 ```
 
-This is a huge improvement! Code size reduction is down from 30 lines to 14, less than half the original size, and faster too. In both cases, you will note that the main loop `.L3` is exactly the same, but the entry and exit code of the function are very much simplified, because the compiler was able to distinguish that the memory pointed by `in` does not overlap with memory pointed by `out`, it was able to simplify the conditions for entering and exiting the main loop.
+This is a huge improvement! Code size reduction is down from 30 lines to 14, less than half the original size. In both cases, you will note that the main loop (`.L4` in the former case, `.L3` in the latter) is exactly the same, but the entry and exit code of the function are very much simplified, because the compiler was able to distinguish that the memory pointed by `in` does not overlap with memory pointed by `out`, it was able to simplify the code by eliminating the scalar loop.
```

Comment:
` simplified, because the` I think this should be a `.` and then to the last sentence add `and remove the associated code that checked if it needed to enter it` or similar. Since you point out before that the entry code is simplified but not why it's simplified.

Reply:
I've added the extra explanation as you requested.